pkg/provisioner/ironic/ironic.go

```diff
@@ -2194,3 +2194,42 @@ func (p *ironicProvisioner) RemoveBMCEventSubscriptionForNode(subscription metal
 	}
 	return operationComplete()
 }
+
+// Fetches the VirtualMedia details of the BareMetalHost and updates DataImage.
+func (p *ironicProvisioner) GetDataImageStatus() (nodeReservation string, nodeLastError string) {
+	// TODO(hroyrh)
+	// Get BareMetalHost VirtualMedia details and handle errors
+
+	node, _ := p.getNode()
+
+	// In case the error node encountered was related to something else
+	// TODO(hroyrh) : Is this check valid ?
```

Comment:
It should be much safer to check for words "attach" and "detach". We may not log the device always.

Reply:
Added check for "attach" and "detach", this should work fine until we have the vmedia.get api.